lib/pbench/test/unit/server/test_upload.py

```diff
@@ -294,10 +308,71 @@ def read(self):
                 data=data_fp,
                 headers=self.gen_headers(pbench_drb_token, "md5sum"),
             )
-        assert response.status_code == HTTPStatus.INTERNAL_SERVER_ERROR
-        assert response.json.get("message").startswith(
-            "Internal Pbench Server Error: log reference "
-        )
+        assert response.status_code == http_status
+        assert response.json.get("message").startswith(message)
+
+    @pytest.mark.parametrize(
+        "error,http_status,message",
+        (
+            (errno.ENOSPC, HTTPStatus.REQUEST_ENTITY_TOO_LARGE, "Out of space on "),
+            (
+                errno.ENFILE,
+                HTTPStatus.INTERNAL_SERVER_ERROR,
+                "Internal Pbench Server Error:",
+            ),
+            (None, HTTPStatus.INTERNAL_SERVER_ERROR, "Internal Pbench Server Error:"),
+        ),
+    )
+    def test_md5_failure(
+        self,
+        monkeypatch,
+        client,
+        pbench_drb_token,
+        server_config,
+        tarball,
+        error,
+        http_status,
+        message,
+    ):
+        """Test handling of errors from MD5 file creation.
+
+        The intake code reports errno.ENOSPC with 413/REQUEST_ENTITY_TOO_LARGE,
+        but other file create errors are reported as 500/INTERNAL_SERVER_ERROR.
+        """
+        path: Optional[Path] = None
+
+        def nogood_write(
+            self, data: str, encoding: str = None, errors: str = None
+        ) -> int:
+            nonlocal path
+            path = self
+            if error:
+                e = OSError(error, "something went badly")
+            else:
+                e = Exception("Nobody expects the Spanish Exception")
+            raise e
+
+        real_unlink = Path.unlink
+        unlinks = []
+
+        def record_unlink(self, missing_ok: bool = False):
+            unlinks.append(self.name)
+            real_unlink(self, missing_ok=missing_ok)
```

Comment:
Since this function doesn't actually use `missing_ok`, you could just write this as a general pass-through (and avoid problems with parameter names, defaults, and so forth):
```python
        def record_unlink(self, **kwargs):
            unlinks.append(self.name)
            real_unlink(self, **kwargs)
```

Reply:
This was a copy from the earlier test, where, intriguingly, I found some odd behavior when I tried to check `missing_ok` out of `kwargs`. It's not worth playing with it.